// storybook-addon-react-router-v6/dist/esm/hooks/useNavigationEventBuilder.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"]; if (null != _i) { var _s, _e, _x, _r, _arr = [], _n = !0, _d = !1; try { if (_x = (_i = _i.call(arr)).next, 0 === i) { if (Object(_i) !== _i) return; _n = !1; } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0) { ; } } catch (err) { _d = !0, _e = err; } finally { try { if (!_n && null != _i["return"] && (_r = _i["return"](), Object(_r) !== _r)) return; } finally { if (_d) throw _e; } } return _arr; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import { EVENTS } from "../constants";
import { useLocation, useNavigationType, useParams, useSearchParams } from "react-router-dom";
import { useCurrentUrl } from "./useCurrentUrl";
import { useDeepRouteMatches } from "./useDeepRouteMatches";
import { useRef } from "react";
export var useNavigationEventBuilder = function useNavigationEventBuilder() {
  var eventCount = useRef(0);
  var location = useLocation();
  var params = useParams();
  var _useSearchParams = useSearchParams(),
    _useSearchParams2 = _slicedToArray(_useSearchParams, 1),
    search = _useSearchParams2[0];
  var navigationType = useNavigationType();
  var matches = useDeepRouteMatches();
  var searchParams = {};
  search.forEach(function (value, key) {
    searchParams[key] = value;
  });
  var currentUrl = useCurrentUrl();
  var matchesData = matches.map(function (routeMatch) {
    return [routeMatch.route.path, routeMatch.params];
  });
  return function (eventName) {
    switch (eventName) {
      case EVENTS.STORY_LOADED:
        {
          var eventData = {
            url: currentUrl,
            path: location.pathname,
            routeParams: params,
            searchParams: searchParams,
            routeMatches: matchesData,
            hash: location.hash,
            routeState: location.state
          };
          return {
            key: "".concat(EVENTS.STORY_LOADED, "_").concat(eventCount.current++),
            type: EVENTS.STORY_LOADED,
            data: eventData
          };
        }
      case EVENTS.NAVIGATION:
        {
          var _eventData = {
            url: currentUrl,
            path: location.pathname,
            routeParams: params,
            searchParams: searchParams,
            hash: location.hash,
            routeState: location.state,
            routeMatches: matchesData,
            navigationType: navigationType
          };
          return {
            key: "".concat(EVENTS.NAVIGATION, "_").concat(eventCount.current++),
            type: EVENTS.NAVIGATION,
            data: _eventData
          };
        }
      case EVENTS.ROUTE_MATCHES:
        {
          return {
            key: "".concat(EVENTS.ROUTE_MATCHES, "_").concat(eventCount.current++),
            type: EVENTS.ROUTE_MATCHES,
            data: {
              matches: matchesData
            }
          };
        }
    }
  };
};
