// storybook-addon-react-router-v6/dist/cjs/components/PanelContent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.humanReadableEventNames = exports.Wrapper = exports.PatchedScrollArea = exports.PanelContent = void 0;
var _react = _interopRequireWildcard(require("react"));
var _theming = require("@storybook/theming");
var _constants = require("../constants");
var _components = require("@storybook/components");
var _RouterEventDisplayWrapper = require("./RouterEventDisplayWrapper");
var _ThemedInspector = require("./ThemedInspector");
var _InspectorContainer = require("./InspectorContainer");
var _humanReadableEventNa;
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
var PatchedScrollArea = _components.ScrollArea;
exports.PatchedScrollArea = PatchedScrollArea;
var PanelContent = function PanelContent(_ref) {
  var navigationEvents = _ref.navigationEvents,
    onClear = _ref.onClear;
  return /*#__PURE__*/_react["default"].createElement(_react.Fragment, null, /*#__PURE__*/_react["default"].createElement(Wrapper, {
    title: "reactRouterLogger"
  }, navigationEvents.map(function (event) {
    return /*#__PURE__*/_react["default"].createElement(_RouterEventDisplayWrapper.RouterEventDisplayWrapper, {
      key: event.key
    }, /*#__PURE__*/_react["default"].createElement(_InspectorContainer.InspectorContainer, null, /*#__PURE__*/_react["default"].createElement(_ThemedInspector.ThemedInspector, {
      name: humanReadableEventNames[event.type],
      data: event.data,
      showNonenumerable: false,
      sortObjectKeys: false,
      expandPaths: ['$.routeParams', '$.searchParams', '$.routeMatches.*', '$.routeMatches.*.*', '$.matches', '$.matches.*', '$.matches.*.*']
    })));
  })), /*#__PURE__*/_react["default"].createElement(_components.ActionBar, {
    actionItems: [{
      title: 'Clear',
      onClick: onClear
    }]
  }));
};
exports.PanelContent = PanelContent;
var humanReadableEventNames = (_humanReadableEventNa = {}, _defineProperty(_humanReadableEventNa, _constants.EVENTS.NAVIGATION, "Navigate"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.STORY_LOADED, "Story rendered"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.ROUTE_MATCHES, "New route matches"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.ACTION_INVOKED, "Action invoked"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.ACTION_SETTLED, "Action settled"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.LOADER_INVOKED, "Loader invoked"), _defineProperty(_humanReadableEventNa, _constants.EVENTS.LOADER_SETTLED, "Loader settled"), _humanReadableEventNa);
exports.humanReadableEventNames = humanReadableEventNames;
var Wrapper = (0, _theming.styled)(function (_ref2) {
  var children = _ref2.children,
    className = _ref2.className;
  return /*#__PURE__*/_react["default"].createElement(PatchedScrollArea, {
    horizontal: true,
    vertical: true,
    className: className
  }, children);
})({
  margin: 0,
  padding: '10px 5px 20px'
});
exports.Wrapper = Wrapper;
Wrapper.displayName = "Wrapper";
