// storybook-addon-react-router-v6/dist/cjs/stories/StoryRouteTree/Basics.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.SpecificPath = exports.SearchParams = exports.RouteParams = exports.RenderChildrenWithStoryArgs = exports.RenderChildren = exports.OutletJSX = exports.OutletConfigObject = exports.MatchesHandlesInsideOutlet = exports.MatchesHandles = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactRouterDom = require("react-router-dom");
var _withRouter = require("../../withRouter");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"]; if (null != _i) { var _s, _e, _x, _r, _arr = [], _n = !0, _d = !1; try { if (_x = (_i = _i.call(arr)).next, 0 === i) { if (Object(_i) !== _i) return; _n = !1; } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0) { ; } } catch (err) { _d = !0, _e = err; } finally { try { if (!_n && null != _i["return"] && (_r = _i["return"](), Object(_r) !== _r)) return; } finally { if (_d) throw _e; } } return _arr; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var DummyComponent = function DummyComponent(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, children);
};
var _default = {
  decorators: [_withRouter.withRouter]
};
exports["default"] = _default;
var RenderChildren = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement("h1", null, "Hi");
  }
};
exports.RenderChildren = RenderChildren;
var RenderChildrenWithStoryArgs = {
  render: function render(_ref2) {
    var id = _ref2.id;
    return /*#__PURE__*/_react["default"].createElement("h1", null, id);
  },
  args: {
    id: "42"
  }
};
exports.RenderChildrenWithStoryArgs = RenderChildrenWithStoryArgs;
function ShowPath() {
  var location = (0, _reactRouterDom.useLocation)();
  return /*#__PURE__*/_react["default"].createElement("p", null, location.pathname);
}
var SpecificPath = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(ShowPath, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/foo'
    }
  }
};
exports.SpecificPath = SpecificPath;
function ShowRouteParams() {
  var routeParams = (0, _reactRouterDom.useParams)();
  return /*#__PURE__*/_react["default"].createElement("p", null, JSON.stringify(routeParams));
}
var RouteParams = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(ShowRouteParams, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/book/:id',
      routeParams: {
        id: '42'
      }
    }
  }
};
exports.RouteParams = RouteParams;
function ShowSearchParams() {
  var _useSearchParams = (0, _reactRouterDom.useSearchParams)(),
    _useSearchParams2 = _slicedToArray(_useSearchParams, 1),
    searchParams = _useSearchParams2[0];
  return /*#__PURE__*/_react["default"].createElement("p", null, JSON.stringify(Object.fromEntries(searchParams.entries())));
}
var SearchParams = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(ShowSearchParams, null);
  },
  parameters: {
    reactRouter: {
      searchParams: {
        page: '42'
      }
    }
  }
};
exports.SearchParams = SearchParams;
function ShowHandles() {
  var matches = (0, _reactRouterDom.useMatches)();
  return /*#__PURE__*/_react["default"].createElement("p", null, JSON.stringify(matches.map(function (m) {
    return m.handle;
  })));
}
var MatchesHandles = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(ShowHandles, null);
  },
  parameters: {
    reactRouter: {
      routeHandle: "Hi"
    }
  }
};
exports.MatchesHandles = MatchesHandles;
var MatchesHandlesInsideOutlet = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(ShowHandles, null);
  },
  parameters: {
    reactRouter: {
      routeHandle: "Hi",
      outlet: {
        handle: "Yall",
        element: /*#__PURE__*/_react["default"].createElement(ShowHandles, null)
      }
    }
  }
};
exports.MatchesHandlesInsideOutlet = MatchesHandlesInsideOutlet;
var OutletJSX = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Outlet, null);
  },
  parameters: {
    reactRouter: {
      outlet: /*#__PURE__*/_react["default"].createElement("h1", null, "I'm an outlet")
    }
  }
};
exports.OutletJSX = OutletJSX;
var OutletConfigObject = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Outlet, null);
  },
  parameters: {
    reactRouter: {
      outlet: {
        element: /*#__PURE__*/_react["default"].createElement("h1", null, "I'm an outlet defined with a config object")
      }
    }
  }
};
exports.OutletConfigObject = OutletConfigObject;
