// storybook-addon-react-router-v6/dist/cjs/stories/StoryRouteTree/Nesting.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.MatchingRoute = exports.MatchingNestedRoute = exports.IndexAtRootWithStoryArgs = exports.IndexAtRoot = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactRouterDom = require("react-router-dom");
var _withRouter = require("../../withRouter");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
var _default = {
  decorators: [_withRouter.withRouter]
};
exports["default"] = _default;
function NestedRoutes() {
  return /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Routes, null, /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    index: true,
    element: /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Link, {
      to: "13"
    }, "Navigate to listing")
  }), /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    path: ":id/*",
    element: /*#__PURE__*/_react["default"].createElement(Listing, null)
  }));
}
function Listing() {
  var params = (0, _reactRouterDom.useParams)();
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("h1", null, "Listing id: ", params.id), /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Routes, null, /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    index: true,
    element: /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Link, {
      to: "FixedId"
    }, "Navigate to details")
  }), /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    path: ":subId",
    element: /*#__PURE__*/_react["default"].createElement(SubListingDetailPage, null)
  })));
}
function SubListingDetailPage() {
  var params = (0, _reactRouterDom.useParams)();
  return /*#__PURE__*/_react["default"].createElement("h2", null, "Details id: ", params.subId);
}
var IndexAtRoot = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing'
    }
  }
};
exports.IndexAtRoot = IndexAtRoot;
function NestedRoutesWithProp(_ref) {
  var _ref$foo = _ref.foo,
    foo = _ref$foo === void 0 ? 1 : _ref$foo;
  return /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Routes, null, /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    index: true,
    element: /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("h1", null, "Story arg : ", foo), /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Link, {
      to: "".concat(foo)
    }, "Navigate to listing"))
  }), /*#__PURE__*/_react["default"].createElement(_reactRouterDom.Route, {
    path: ":id/*",
    element: /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("h2", null, "Story arg : ", foo), /*#__PURE__*/_react["default"].createElement(Listing, null))
  }));
}
var IndexAtRootWithStoryArgs = {
  render: function render(_ref2) {
    var foo = _ref2.foo;
    return /*#__PURE__*/_react["default"].createElement(NestedRoutesWithProp, {
      foo: foo
    });
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing'
    }
  },
  args: {
    foo: 42
  }
};
exports.IndexAtRootWithStoryArgs = IndexAtRootWithStoryArgs;
var MatchingRoute = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing/13'
    }
  }
};
exports.MatchingRoute = MatchingRoute;
var MatchingNestedRoute = {
  render: function render() {
    return /*#__PURE__*/_react["default"].createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing/13/37'
    }
  }
};
exports.MatchingNestedRoute = MatchingNestedRoute;
