// storybook-addon-react-router-v6/dist/esm/components/StoryRouter.js
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"]; if (null != _i) { var _s, _e, _x, _r, _arr = [], _n = !0, _d = !1; try { if (_x = (_i = _i.call(arr)).next, 0 === i) { if (Object(_i) !== _i) return; _n = !1; } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0) { ; } } catch (err) { _d = !0, _e = err; } finally { try { if (!_n && null != _i["return"] && (_r = _i["return"](), Object(_r) !== _r)) return; } finally { if (_d) throw _e; } } return _arr; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
import React, { useLayoutEffect, useRef, useState } from 'react';
import { createMemoryRouter, createRoutesFromElements, generatePath, RouterProvider } from "react-router-dom";
import { addons } from "@storybook/addons";
import { STORY_ARGS_UPDATED } from "@storybook/core-events";
export var StoryRouter = function StoryRouter(_ref) {
  var children = _ref.children,
    routePath = _ref.routePath,
    routeParams = _ref.routeParams,
    searchParams = _ref.searchParams,
    routeState = _ref.routeState,
    userBrowserPath = _ref.browserPath,
    hydrationData = _ref.hydrationData;
  var channel = addons.getChannel();
  var _useState = useState(),
    _useState2 = _slicedToArray(_useState, 2),
    router = _useState2[0],
    setRouter = _useState2[1];
  var _useState3 = useState(0),
    _useState4 = _slicedToArray(_useState3, 2),
    storyMutations = _useState4[0],
    setStoryMutations = _useState4[1];
  var routeStateRef = useRef();
  channel.on(STORY_ARGS_UPDATED, function () {
    setStoryMutations(function (prev) {
      return prev + 1;
    });
  });
  useLayoutEffect(function () {
    var generatedPath = generatePath(routePath || '', routeParams);
    var queryString = new URLSearchParams(searchParams).toString();
    var search = queryString.length > 0 ? "?".concat(queryString) : '';
    var initialEntry = {
      search: search,
      state: routeState
    };
    if (userBrowserPath !== undefined) initialEntry['pathname'] = userBrowserPath;
    if (userBrowserPath === undefined && generatedPath !== '') initialEntry['pathname'] = generatedPath;
    if (routeStateRef.current !== undefined) {
      Object.assign(initialEntry, routeStateRef.current);
    }
    var routes = createRoutesFromElements(children);
    var memoryRouter = createMemoryRouter(routes, {
      initialEntries: [initialEntry],
      hydrationData: hydrationData
    });
    memoryRouter.subscribe(function (nextState) {
      routeStateRef.current = nextState.location;
    });
    setRouter(memoryRouter);
  }, [storyMutations]);
  if (router !== undefined && module && module.hot && module.hot.dispose) {
    module.hot.dispose(function () {
      return router.dispose();
    });
  }
  if (router === undefined) return null;
  return /*#__PURE__*/React.createElement(RouterProvider, {
    router: router,
    fallbackElement: /*#__PURE__*/React.createElement(Fallback, null)
  });
};
export function Fallback() {
  return /*#__PURE__*/React.createElement("p", null, "Performing initial data load");
}
