// storybook-addon-react-router-v6/dist/esm/stories/StoryRouteTree/Nesting.stories.js
import React from "react";
import { Link, Route, Routes, useParams } from "react-router-dom";
import { withRouter } from "../../withRouter";
export default {
  decorators: [withRouter]
};
function NestedRoutes() {
  return /*#__PURE__*/React.createElement(Routes, null, /*#__PURE__*/React.createElement(Route, {
    index: true,
    element: /*#__PURE__*/React.createElement(Link, {
      to: "13"
    }, "Navigate to listing")
  }), /*#__PURE__*/React.createElement(Route, {
    path: ":id/*",
    element: /*#__PURE__*/React.createElement(Listing, null)
  }));
}
function Listing() {
  var params = useParams();
  return /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", null, "Listing id: ", params.id), /*#__PURE__*/React.createElement(Routes, null, /*#__PURE__*/React.createElement(Route, {
    index: true,
    element: /*#__PURE__*/React.createElement(Link, {
      to: "FixedId"
    }, "Navigate to details")
  }), /*#__PURE__*/React.createElement(Route, {
    path: ":subId",
    element: /*#__PURE__*/React.createElement(SubListingDetailPage, null)
  })));
}
function SubListingDetailPage() {
  var params = useParams();
  return /*#__PURE__*/React.createElement("h2", null, "Details id: ", params.subId);
}
export var IndexAtRoot = {
  render: function render() {
    return /*#__PURE__*/React.createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing'
    }
  }
};
function NestedRoutesWithProp(_ref) {
  var _ref$foo = _ref.foo,
    foo = _ref$foo === void 0 ? 1 : _ref$foo;
  return /*#__PURE__*/React.createElement(Routes, null, /*#__PURE__*/React.createElement(Route, {
    index: true,
    element: /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", null, "Story arg : ", foo), /*#__PURE__*/React.createElement(Link, {
      to: "".concat(foo)
    }, "Navigate to listing"))
  }), /*#__PURE__*/React.createElement(Route, {
    path: ":id/*",
    element: /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h2", null, "Story arg : ", foo), /*#__PURE__*/React.createElement(Listing, null))
  }));
}
export var IndexAtRootWithStoryArgs = {
  render: function render(_ref2) {
    var foo = _ref2.foo;
    return /*#__PURE__*/React.createElement(NestedRoutesWithProp, {
      foo: foo
    });
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing'
    }
  },
  args: {
    foo: 42
  }
};
export var MatchingRoute = {
  render: function render() {
    return /*#__PURE__*/React.createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing/13'
    }
  }
};
export var MatchingNestedRoute = {
  render: function render() {
    return /*#__PURE__*/React.createElement(NestedRoutes, null);
  },
  parameters: {
    reactRouter: {
      routePath: '/listing/*',
      browserPath: '/listing/13/37'
    }
  }
};
